cognite_toolkit/cdf_tk/load/_resource_loaders.py

```diff
@@ -232,11 +223,30 @@ def load_resource(
                             for ext_id in ids
                         ]
 
-        if not is_resource_scoped and self.target_scopes == "resource_scoped_only":
-            # If a group has no resource scoped capabilities, we skip it.
+        return raw
+
+    def load_resource(
+        self, filepath: Path, ToolGlobals: CDFToolConfig, skip_validation: bool
+    ) -> GroupWrite | GroupWriteList | None:
+        raw = load_yaml_inject_variables(filepath, ToolGlobals.environment_variables(), required_return_type="dict")
+
+        is_resource_scoped = any(
+            any(scope_name in capability.get(acl, {}).get("scope", {}) for scope_name in self.resource_scope_names)
+            for capability in raw.get("capabilities", [])
+            for acl in capability
+        )
+
+        if self.target_scopes == "all_scoped_only":
+            if not is_resource_scoped:
+                return GroupWrite.load(raw)
             return None
```

Comment:
Nested ifs are typically hard to read
```suggestion
        if self.target_scopes == "all_scoped_only" and is_resource_scoped:
            return None
```